src/Silex/CallbackResolver.php

```diff
@@ -33,35 +33,64 @@ public function __construct(Container $app)
      */
     public function isValid($name)
     {
-        return is_string($name) && preg_match(static::SERVICE_PATTERN, $name);
+        return is_string($name) && (
+            preg_match(static::SERVICE_PATTERN, $name) ||
+            isset($this->app[$name])
+        );
+    }
+
+    /**
+     * Throws an InvalidArgumentException for the service not existing.
+     *
+     * @param string $name
+     *
+     * @throws \InvalidArgumentException
+     */
+    protected function throwServiceNotExists($service)
```

Comment:
Creating a `private` method sounds better to me (should be moved at the end of the file).